### PR TITLE
Dev2

### PR DESCRIPTION
**Pull request by RokasAlechnavicius**
Login form, Login with call to backend service, form validation, JWT interceptor, Auth guard

### CODE REVIEW

File: src/app/app.component.spec.ts [hunk 1]
```diff
@@ -22,6 +23,8 @@ import { FormsModule, ReactiveFormsModule } from '@angular/forms';
 import { MatInputModule } from '@angular/material/input';
 import { MatSelectModule } from '@angular/material/select';
 import { AlertComponent } from './alert/alert.component';
+import { LoginComponent } from './login/login.component';
+import {HttpClientTestingModule} from '@angular/common/http/testing';
```
Comment (matasgraj) [inline src/app/app.component.spec.ts]: Formatting :)

File: src/app/guards/auth.guard.spec.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+describe('AuthGuard', () => {
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            providers: [AuthGuard],
+            imports: [RouterTestingModule, HttpClientTestingModule]
+        });
+    });
+
+    it('should ...', inject([AuthGuard], (guard: AuthGuard) => {
```
Comment (matasgraj) [inline src/app/guards/auth.guard.spec.ts]: Should do what?

File: src/app/helpers/jwt.interceptor.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+})
+export class JwtInterceptor implements HttpInterceptor {
+    constructor(private userAuthenticationService: UserAuthenticationService) {}
+
+    intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+        const currentUser = this.userAuthenticationService.currentUserValue;
+        if (currentUser && currentUser.token) {
+            request = request.clone({
+                setHeaders: {
+                    Authorization: 'Bearer ${currentUser.token'
```
Comment (matasgraj) [inline src/app/helpers/jwt.interceptor.ts]: Shouldn't this end with "}"? `Authorization: 'Bearer ${currentUser.token}'`

File: src/app/login/login.component.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        private formBuilder: FormBuilder,
+        private route: ActivatedRoute,
+        private router: Router,
+        private userAuthenticationService: UserAuthenticationService,
+        private alertService: AlertService
+    ) {}
+
+    ngOnInit() {
+        this.createForm();
+
```
Comment (matasgraj) [inline src/app/login/login.component.ts]: Why have you started to put spaces between lines? In this case, no need for that

**Review by matasgraj — COMMENTED** [review]
Good job with login-in handling 👍  there is some minor fixes that could be done until the merge

**Review by matasgraj — APPROVED** [review]

**Review by TomasBrezinskas — APPROVED** [review]

**Pull request merged**